app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java

```diff
@@ -260,15 +263,52 @@ private void setupTimelinePreferences() {
         filter = preferences.getBoolean("tabFilterHomeBoosts", true);
         filterRemoveReblogs = kind == Kind.HOME && !filter;
 
-        String regexFilter = preferences.getString("tabFilterRegex", "");
-        filterRemoveRegex = (kind == Kind.HOME
-                || kind == Kind.PUBLIC_LOCAL
-                || kind == Kind.PUBLIC_FEDERATED)
-                && !regexFilter.isEmpty();
+        mastodonApi.getFilters().enqueue(new Callback<List<Filter>>() {
+            @Override
+            public void onResponse(Call<List<Filter>> call, Response<List<Filter>> response) {
+                applyFilters(response.body());
+            }
+
+            @Override
+            public void onFailure(Call<List<Filter>> call, Throwable t) {
+                Log.e(TAG, "Error getting filters from server");
+            }
+        });
+    }
+
+    private static boolean filterContextMatchesKind(Kind kind, List<String> filterContext) {
+        // home, notifications, public, thread
+        switch(kind) {
+            case HOME:
+                return filterContext.contains("home");
+            case PUBLIC_FEDERATED:
+            case PUBLIC_LOCAL:
+            case USER:
```

Comment:
I think `USER` and `USER_WITH_REPLIES` need to be removed, since the web UI doesn't apply filters to the account UI

Reply:
yes `USER` and `USER_WITH_REPLIES` should be removed